### PR TITLE
Prevent dotnet-watch script injection staleness

### LINKED ISSUE #27548: dotnet watch run script injection fails to update etag
I've been building a Blazor WebAssembly app via `dotnet watch run` for a while, and noticed that when the auto-opened browser tab first loads, it fails to connect to the auto-refresh websocket endpoint. The error in the browser dev console is this:

```
WebSocket connection to 'ws://127.0.0.1:54999/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED
```

I can reload the page and it still fails to connect. But if I hard-reload (ctrl+shift+r), it then actually succeeds in connecting (but to a different port number).

The reason for this seems pretty clear. The browser is asking for the `index.html` content with an `if-modified-since: <etag>` header. The server is responding `304 Not Modified` **which is a lie** because if you hard-reload, then you will get back different content (different websocket port number) but with the same etag header in the response.

**Resolution**

`dotnet watch run` needs to either:

 * Inject script content in such a way that is reflected in the etag
 * Or, if that's not possible, modify the response headers to include `cache-control: no-cache` or similar. I think that would be good enough given the use case.

### PR DESCRIPTION
**Pull request by pranavkm**
The browser refresh mechansim used by dotnet-watch and VS modifies HTML content. The modified content includes
a script block that has a WebSocket url that changes for each new execution of `dotnet watch run` (not rebuilds, but watch itself).
HTML content can come from views or static html files on disk. For the latter, ASP.NET Core participates in browser caching by sending (and invalidating) etag headers.

One way to fix this problem is remove or modify the etag headers. The risk here is that might cause differences in behavior in development users may come to rely on that are unavailable in production. This change instead modifies the HTML content so the output is always consistent and consequently safe to cache. The dynamic content is served separately by the injected middleware.

This change fixes the issue of multiple instances of dotnet-watch. While this issue may crop up if you alternate between `dotnet run` and `dotnet watch run` but we haven't seen this being an issue as yet.

Fixes https://github.com/dotnet/aspnetcore/issues/27548

### Summary

Running `dotnet watch run` multiple times in Blazor WASM apps (or any app that serves static html files) can produce console errors and prevent the browser refresh feature from working. Given that we've been telling our users to use `dotnet watch run` as their primary way to work outside of VS, it's likely more users would run in this.

### Customer impact
A hard browser refresh (Ctrl + R) is needed to get the refresh behavior to work. 

### Regression
No. This has existed since the feature was introduced we did not get reports of it

### Risk
Low. The fix is isolated to dotnet-watch and VS's browser refresh mechanism which is in preview. The change was tested locally, but if there's a regression or if the change interferes with user's workflow, users have the ability to disable this feature.

### CODE REVIEW

File: src/Tools/dotnet-watch/BrowserRefresh/src/BrowserScriptMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            _contentLength = _scriptBytes.Length.ToString(CultureInfo.InvariantCulture);
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            context.Response.Headers["Cache-Control"] = "no-store";
+            context.Response.Headers["Content-Length"] = _contentLength;
+            context.Response.Headers["Content-Type"] = "application/javascript; charset=utf-8";
+
+            await context.Response.Body.WriteAsync(_scriptBytes.AsMemory(), context.RequestAborted);
```
Comment (Tratcher) [inline src/Tools/dotnet-watch/BrowserRefresh/src/BrowserScriptMiddleware.cs:39]: No async/await required.

**Review by Tratcher — APPROVED** [review]

**Comment by mkArtakMSFT** [comment]
Approved by Tactics!

File: src/Tools/dotnet-watch/BrowserRefresh/src/WebSocketScriptInjection.cs [hunk 2]
```diff
@@ -13,22 +13,18 @@ namespace Microsoft.AspNetCore.Watch.BrowserRefresh
     /// Helper class that handles the HTML injection into
     /// a string or byte array.
     /// </summary>
-    public class WebSocketScriptInjection
+    public static class WebSocketScriptInjection
     {
         private const string BodyMarker = "</body>";
+        internal const string WebSocketScriptUrl = "/_aspnetcore-dev-tools/browser-refresh.js";
```
Comment (SteveSandersonMS) [inline src/Tools/dotnet-watch/BrowserRefresh/src/WebSocketScriptInjection.cs:19]: Do you think we should consider standardizing on `/_framework/*` for system-generated URLs? I know that means we have to make sure we don't clash within that namespace, but it means there's only one top-level space that developers have to account for if they are trying to make sure framework-generated stuff is allowed through whatever other router customizations they are doing.

**Review by SteveSandersonMS — APPROVED** [review]
I like this approach!

**Comment by mkArtakMSFT** [comment]
@pranavkm let me know when this is ready to merge.

File: src/Tools/dotnet-watch/BrowserRefresh/src/WebSocketScriptInjection.cs (hunk 2 above)
Comment (pranavkm) [reply src/Tools/dotnet-watch/BrowserRefresh/src/WebSocketScriptInjection.cs]: That's a great idea and a good way to to ensure user and framework endpoints don't stomp on each other. I've updated my PR to include this feedback.

**Pull request merged**